src/Compiler/Compiler/OtherHelpersAndHandlers/MonoCecilAssembliesInspector/MonoCecilAssembliesInspectorImpl.cs

```diff
@@ -0,0 +1,781 @@
+        {
+            // Fix the namespace:
+            if (namespaceName.StartsWith(Using, StringComparison.CurrentCultureIgnoreCase))
+            {
+                namespaceName = namespaceName.Substring(Using.Length);
+            }
+            else if (namespaceName.StartsWith(ClrNamespace, StringComparison.CurrentCultureIgnoreCase))
+            {
+                GettingInformationAboutXamlTypes.ParseClrNamespaceDeclaration(namespaceName, out var ns,
+                    out var assemblyNameIfAny);
+                namespaceName = ns;
+                GettingInformationAboutXamlTypes.FixNamespaceForCompatibility(ref assemblyNameIfAny, ref namespaceName);
+            }
+
+            if (namespaceName.StartsWith(GlobalPrefix,
+                    StringComparison
+                        .CurrentCultureIgnoreCase)) // Note: normally in XAML there is no "global::", but we may enter this method passing a C#-style namespace (cf. section that handles Binding in "GeneratingCSharpCode.cs")
+                namespaceName = namespaceName.Substring(GlobalPrefix.Length);
+
+            // Handle special cases:
+            if (localTypeName == StaticRes) localTypeName = StaticResExtension;
+
+            // Generate string representing the type:
+            var fullTypeNameWithNamespaceInsideBraces = !string.IsNullOrEmpty(namespaceName)
+                ? "{" + namespaceName + "}" + localTypeName
+                : localTypeName;
+
+            // Start by looking in the cache dictionary:
+            if (_typeNameToType.TryGetValue(fullTypeNameWithNamespaceInsideBraces, out var type)) return type;
+
+            // Look for the type in all loaded assemblies:
+            foreach (var assemblyKeyValuePair in _storage.LoadedAssemblySimpleNameToAssembly)
+            {
+                var assemblySimpleName = assemblyKeyValuePair.Key;
+                var assembly = assemblyKeyValuePair.Value;
+                if (filterAssembliesAndRetainOnlyThoseThatHaveThisName == null
+                    || assemblySimpleName == filterAssembliesAndRetainOnlyThoseThatHaveThisName)
+                {
+                    var namespacesToLookInto = new List<string>();
+
+                    // If the namespace is a XML namespace (eg. "{http://schemas.microsoft.com/winfx/2006/xaml/presentation}"), we should iterate through all the corresponding CLR namespaces:
+                    if (IsNamespaceAnXmlNamespace(namespaceName))
+                        namespacesToLookInto.AddRange(
+                            GetClrNamespacesFromXmlNamespace(assemblySimpleName, namespaceName));
+                    else
+                        namespacesToLookInto.Add(namespaceName);
+
+                    // Search for the type:
+                    foreach (var namespaceToLookInto in namespacesToLookInto)
+                    {
+                        var fullTypeNameToFind = namespaceToLookInto + "." + localTypeName;
```

Comment:
It seems that Assembly.GetType(...) ignore the '.' when the namespace is empty but that is not the case for mono.cecil, so let's just check the namespace
```suggestion
                        var fullTypeNameToFind = string.IsNullOrEmpty(namespaceToLookInto) ? localTypeName : $"{namespaceToLookInto}.{localTypeName}";
```